src/pages/Charcuterie.js

```diff
@@ -279,6 +280,11 @@ export default () => {
         debounceEnabled
         throttleTime={1000}
       />
+      <QuestionHeading>question 12</QuestionHeading>
+      <div>
+        <ResumeUploadBtn />
```

Comment:
I think having the `marginLeft: '9em'` styling here on the parent div would be more appropriate to show how it's used and how you could style it
With the suggestions I have for passing in additional props, I think keeping track of hint and handleChange could be moved here
something like
```
<ResumeUploadBtn 
     hint={hint.match(/[\/\\]([\w\d\s\.\-\(\)]+)$/)[1]} 
     onChange={(e) => setHint(e.target.value)} 
     isValid={hint} 
     errorMsg="Please upload your resume" 
/>
```
and at the top of this file you would have `const [hint, setHint] = ...` etc

Reply:
thanks for the suggestion! I've made the change to pass the props and moved the hint state into this file as well. But I'm not entirely sure what you mean by moving the marginLeft styling here since originally i used that styling so that there is more space between the "Resume" label on the left and the button + hint on the right. I'm not too sure how I would use the margin here in this case?

Reply:
also I wasn't sure how to do the regular expression matching here. since it's possible for the hint to be undefined/null still if the state is undefined initially. I tried it with an initial state of empty string, but then the match result would be null I believe which also causes an error when trying to access the index [1] and I tried using conditionals to only call the match if the hint exists, but it seems like the hint state does not update properly? so I couldn't get it to work and decided to leave the matching in the Upload component still, let me know if that's alright!